src/main/java/com/productdock/library/gateway/book/BookService.java

```diff
@@ -2,43 +2,60 @@
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.jwt.JWTParser;
 import com.productdock.library.gateway.client.CatalogClient;
 import com.productdock.library.gateway.client.InventoryClient;
 import com.productdock.library.gateway.client.RentalClient;
 import lombok.SneakyThrows;
 import org.springframework.stereotype.Service;
 import reactor.core.publisher.Mono;
+import reactor.util.function.Tuple4;
+
+import java.text.ParseException;
+import java.util.List;
 
 @Service
 public record BookService(CatalogClient catalogClient, RentalClient rentalClient,
                           InventoryClient inventoryClient, BookDetailsResponseCombiner bookDetailsResponseCombiner) {
 
+    private static final String CLAIM_EMAIL = "email";
+
     @SneakyThrows
     public JsonNode getBookDetailsById(String bookId, String jwtToken) {
+        var userId = getClaimEmail(jwtToken);
```

Comment:
Rename `getClaimEmail` to`extractUserId` or similar, because `userId` is what you get